tools/pyvmi/libvmi/libvmi_build.py

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python3
+
+
+import os
+from cffi import FFI
+
+CDEF_FILE = 'libvmi_cdef.h'
+
+
+ffi = FFI()
+# set source
+ffi.set_source("_libvmi",
+    """
+    #include <libvmi/libvmi.h>
+    """,
+    libraries=['vmi'])
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+# we read our C definitions from an external file
+# easier to maintain + C syntax highlighting
```

Comment:
I believe it would be more maintainable to use libvmi/libvmi.h directly rather than making a shadow version of it in libvmi_cdef

Reply:
I have been advised the contrary: just pick the APIs that i'm really using in my cffi and copy paste their prototype.
Also, CFFI is not able to fully parse C headers as of today, so a slightest "exotic" C declaration could break `libvmi_build.py`.

But let me try.

Reply:
ping @noxdafox , can i have your opinion on CFFI ?

Reply:
Yep, it fails.